certbot/certbot/_internal/client.py

```diff
@@ -528,6 +534,22 @@ def obtain_and_enroll_certificate(self, domains: List[str], certname: Optional[s
             key.pem, chain,
             self.config)
 
+    def _retry_obtain_certificate(self, error: messages.Error, key: util.Key,
+                                csr: util.CSR, domains: List[str],
+                                ) -> Tuple[bytes, bytes, util.Key, util.CSR]:
+        if self.config.allow_subset_of_names and error.subproblems is not None:
+            failed_domains = [problem.identifier.value for problem in error.subproblems
+                                if problem.identifier is not None]
+            successful_domains = [x for x in domains if x not in failed_domains]
+        if successful_domains != domains and len(successful_domains) != 0:
+            display_util.notify("Unable to obtain a certificate with every requested "
+                "domain. Retrying without: {0}".format(failed_domains))
```

Comment:
We try to use f-strings now I think:

```suggestion
                f"domain. Retrying without: {failed_domains}")
```

Reply:
Done! I'll try to remember this 🙂